src/btree/bt_split.c

```diff
@@ -936,7 +965,22 @@ __split_parent(WT_SESSION_IMPL *session, WT_REF *ref,
 				 */
 				ref_new[j] = NULL;
 			}
-		else if (next_ref->state != WT_REF_SPLIT)
+
+			/*
+			 * We detect append-style workloads to avoid repeatedly
+			 * deepening parts of the tree where no work is being
+			 * done by tracking if we're splitting after the slots
+			 * created by the last split to deepen this parent.
+			 *
+			 * Note the calculation: either or both i and split-last
+			 * might be an unsigned 0, don't decrement either one.
+			 * Because i is a 0-based array offset and split-last is
+			 * an count of entries, compare (i + 1) to split-last.
+			 */
+			if (i + 1 >= parent->pg_intl_deepen_split_last)
```

Comment:
This seems convoluted, given that it is equivalent to `if (i > parent->pg_intl_deepen_split_last)` -- the simpler expression makes intuitive sense to me.  Am I missing some subtlety?